httpx/_exceptions.py

```diff
@@ -100,76 +126,103 @@ class CloseError(NetworkError):
 # Other transport exceptions...
 
 
-class ProxyError(HTTPError):
+class ProxyError(TransportError):
     """
     An error occurred while proxying a request.
     """
 
 
-class ProtocolError(HTTPError):
+class ProtocolError(TransportError):
     """
     A protocol was violated by the server.
     """
 
 
-# HTTP exceptions...
+# Other request exceptions...
 
 
-class DecodingError(HTTPError):
+class DecodingError(RequestError):
     """
     Decoding of the response failed.
     """
 
 
-class HTTPStatusError(HTTPError):
+class TooManyRedirects(RequestError):
     """
-    Response sent an error HTTP status.
+    Too many redirects.
     """
 
-    def __init__(self, *args: typing.Any, response: "Response") -> None:
-        super().__init__(*args)
-        self._request = response.request
-        self.response = response
-
 
-# Redirect exceptions...
+class RequestBodyUnavailable(RequestError):
+    """
+    Had to send the request again, but the request body was streaming, and is
+    no longer available.
+    """
 
 
-class RedirectError(HTTPError):
+class InvalidURL(RequestError):
     """
-    Base class for HTTP redirect errors.
+    URL was missing a hostname, or was not one of HTTP/HTTPS.
     """
 
 
-class TooManyRedirects(RedirectError):
+# Client errors
+
+
+class HTTPStatusError(Exception):
     """
-    Too many redirects.
+    Response sent an error HTTP status.
+
+    May be raised when calling `response.raise_for_status()`
     """
 
+    def __init__(
+        self, message: str, *, request: "Request", response: "Response"
+    ) -> None:
+        super().__init__(message)
+        self.request = request
+        self.response = response
 
-class NotRedirectResponse(RedirectError):
+
+class NotRedirectResponse(Exception):
     """
     Response was not a redirect response.
+
+    May be raised if `response.next()` is called without first
+    properly checking `response.is_redirect`.
     """
 
+    def __init__(self, message: str) -> None:
+        super().__init__(message)
+
+
+class CookieConflict(Exception):
+    """
+    Attempted to lookup a cookie by name, but multiple cookies existed.
+
+    Can occur when calling `response.cookies.get(...)`
```

Comment:
(I'm reviewing assuming these docstrings will end up in the public Developer Interface sometime in the near future 😉)

```suggestion
    Can occur when calling `response.cookies.get(...)`.
```